### PR TITLE
update syntax to ioda v2

### LINKED ISSUE #1422: Replace usage of ioda v1 naming with ioda v2 naming in YAML configuration
## Description

complete the yaml changes for v2 IODA syntax replacing the `Variable@Group` which need to be replaced with v2 syntax `Group/Variable`

https://github.com/JCSDA-internal/ioda/issues/379

Try `grep -R "\@" | grep yaml` to find these.
( thanks @srherbener )

Here's an example of a test file using the v1 syntax:

https://github.com/JCSDA-internal/ufo/blob/06943cf6645b0df1700d0a422bdfc5c603cf97ec/test/testinput/unit_tests/filters/qc_met_office_buddy_check_unittests.yaml#L20

Here's an example in the code:

https://github.com/JCSDA-internal/ufo/blob/06943cf6645b0df1700d0a422bdfc5c603cf97ec/src/ufo/variabletransforms/Cal_PStar.cc#L30

There seems to a a fair number of these places, so perhaps this work can be delegated out to others.

### PR DESCRIPTION
**Pull request by BenjaminRuston**
## Description

Provide a detailed description of what this PR does. What problem does it fix? What new capability does it add?

will finish long standing tasks for:  https://github.com/JCSDA-internal/ioda/issues/379

## Issue(s) addressed

Resolves #1422 

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have run the unit tests before creating the PR

### CODE REVIEW

**Comment by BenjaminRuston** [comment]
believe I can fix this error by uploading a new `testoutput` but the error seems strange as would have expected this to work, and suggest the file is writing a field with a name rather than an actual group?

```
PRE-MAIN-INFO BufrParser: Parsing file ./testinput/bufr_sfcshp.bufr
PRE-MAIN-INFO Executing Queries
PRE-MAIN-INFO Building Bufr Data
PRE-MAIN-INFO Exporting Data
PRE-MAIN-INFO Finished [0.022s]
DIFFER : NAME OF GLOBAL ATTRIBUTE : MetaData/dataType : GLOBAL ATTRIBUTE DOESN'T EXIST IN "testoutput/bufr_sfcshp.nc"
DIFFER : NAME OF GLOBAL ATTRIBUTE : dataType@MetaData : GLOBAL ATTRIBUTE DOESN'T EXIST IN testrun/bufr_sfcshp.nc
```

**Comment by PatNichols** [comment]
@BenjaminRuston I did an fgrep on the testoutput files and that particular file is the only one that has that global attribute or even name as something else. Probably something from the bufr file?

**Comment by BenjaminRuston** [comment]
yes this file in the `testoutput` is not defined correctly:
```
python3
>>> import h5py
>>> f = h5py.File('bufr_sfcshp.nc', 'r')
>>> f.keys()
<KeysViewHDF5 ['Location', 'MetaData', 'ObsValue']>
>>> f['MetaData']
<HDF5 group "/MetaData" (5 members)>
>>> f['MetaData'].keys()
<KeysViewHDF5 ['datetime', 'depthBelowWaterSurface', 'latitude', 'longitude', 'stationIdentification']>
>>> f.attrs.keys()
<KeysViewHDF5 ['_ioda_layout', '_ioda_layout_version', 'dataType@MetaData']>
>>> f.attrs['dataType@MetaData']
array(['Moored Buoy'], dtype=object)
```

yes the `dataType@MetaData` should not be a global attribute key

File: test/testinput/bufr_sfcshp.yaml [hunk 1]
```diff
@@ -46,7 +46,7 @@ observations:
       obsdataout: "./testrun/bufr_sfcshp.nc"
 
       globals:
-        - name: "dataType@MetaData"
+        - name: "platformLongDescription"
```
Comment (BenjaminRuston) [inline test/testinput/bufr_sfcshp.yaml:49]: @emilyhcliu this was adding a key `dataType@MetaData` to the global attributes, not adding anything to the `MetaData` group

in the [conventions](https://github.com/JCSDA-internal/ioda/blob/b27373f084f63d7565338f13588118a73787890d/share/ioda/yaml/validation/ObsSpace.yaml#L52) there are many options `long_name`, etc chose this one as think it is appropriate.  Confident this is not being used for anything so do like this name and have been trying to use it in other converters

**Comment by BenjaminRuston** [comment]
this is ready for review if anyone would like to look it over

**Review by PatNichols — APPROVED** [review]
Thanks!

**Review by fabiolrdiniz — APPROVED** [review]
Thanks, @BenjaminRuston!

**Pull request merged**